src/target/renesas.c

```diff
@@ -511,6 +517,10 @@ static bool renesas_rv40_flash_erase(target_flash_s *f, target_addr_t addr, size
 	/* code flash or data flash operation */
 	const bool code_flash = addr < RENESAS_CF_END;
 
+	/* Erasing Option-Setting flash memory is not allowed, only write is supported */
+	if (RENESAS_OF_START <= addr && addr <= (RENESAS_OF_START + RENESAS_OF_SIZE))
+		return true;
```

Comment:
if erasing is not allowed why is there a flash_erase method tied to it, we should omit it, I don't know if that's currently allowed, but It should if not, or have a stub, that said retaining the sanity check might still be good

Reply:
not allowed to not add erase handler in driver, it will crash

> #0  0x0000000000000000 in ?? ()
> #1  0x000055555557a732 in target_flash_erase (t=0x5555555a9630, addr=16818432, len=64) at target/target_flash.c:149
> #2  0x0000555555565320 in handle_v_packet (plen=29, packet=0x5555555a3630 <pbuf> "vFlashErase:0100a100,00000040") at gdb_main.c:624
> #3  gdb_main_loop (tc=<optimized out>, pbuf=0x5555555a3630 <pbuf> "vFlashErase:0100a100,00000040", pbuf_size=<optimized out>, size=29, in_syscall=<optimized out>) at gdb_main.c:339
> #4  0x000055555555c04b in bmp_poll_loop () at main.c:68
> #5  main (argc=<optimized out>, argv=<optimized out>) at main.c:84

Reply:
@dragonmux do we have stub for such situations?

Reply:
We do not - you could modify `target_flash_erase()` to check the pointer for NULL before calling it (treating nullptr as returning `true`).

One note of feedback about the check done in the snippet above, because `RENESAS_OF_START + RENESAS_OF_SIZE` gives an address that is more than the address range of the region, you want the second comparison to just be `addr < `, not `addr <= `